src/timeline/TimelineModel.cpp

```diff
@@ -1934,11 +1934,298 @@ TimelineModel::formatPowerLevelEvent(const QString &id)
+                             utils::replaceEmoji(displayName(affected.at(1))))
+                        .arg(true_affected_rest));
+                } else if (number_of_affected == 2) {
+                    resultingMessage.append(
+                      default_message + QString::fromStdString(" ") +
+                      tr("%1 and %2 can now kick room members.")
+                        .arg(utils::replaceEmoji(displayName(affected.at(0))),
+                             utils::replaceEmoji(displayName(affected.at(1)))));
+                } else if (number_of_affected == 1) {
+                    resultingMessage.append(
+                      default_message + QString::fromStdString(" ") +
+                      tr("%1 can now kick room members.")
+                        .arg(utils::replaceEmoji(displayName(affected.at(0)))));
+                }
+            } else {
+                resultingMessage.append(default_message);
+            }
+        } else {
+            resultingMessage.append(default_message);
+        }
+    }
+
+    if (event->content.redact != prevEvent->content.redact) {
+        auto default_message = tr("%1 has changed the room's redact powerlevel from %2 to %3.")
+                                 .arg(sender_name)
+                                 .arg(prevEvent->content.redact)
+                                 .arg(event->content.redact);
+
+        // We only calculate affected users if we change to a level above the default users PL
+        // to not accidentally have a DoS vector
+        if (event->content.redact > default_powerlevel) {
+            auto [affected, number_of_affected] = calc_affected(event->content.kick);
+
+            if (number_of_affected != 0) {
+                auto true_affected_rest = number_of_affected - affected.size();
+                if (number_of_affected > 2) {
+                    resultingMessage.append(
+                      default_message + QString::fromStdString(" ") +
+                      tr("%1 and %2 and %n other(s) can now redact room messages.")
+                        .arg(utils::replaceEmoji(displayName(affected.at(0))),
+                             utils::replaceEmoji(displayName(affected.at(1))))
+                        .arg(true_affected_rest));
+                } else if (number_of_affected == 2) {
+                    resultingMessage.append(
+                      default_message + QString::fromStdString(" ") +
+                      tr("%1 and %2 can now redact room messages.")
+                        .arg(utils::replaceEmoji(displayName(affected.at(0))),
+                             utils::replaceEmoji(displayName(affected.at(1)))));
+                } else if (number_of_affected == 1) {
+                    resultingMessage.append(
+                      default_message + QString::fromStdString(" ") +
+                      tr("%1 can now redact room messages.")
+                        .arg(utils::replaceEmoji(displayName(affected.at(0)))));
```

Comment:
I'd probably cut those down to 2 cases, "%1 can now redact messages." and "%n additional users can now redact messages".